planner/memo/expr_iterator.go

```diff
@@ -102,6 +107,10 @@ func (iter *ExprIter) Matched() bool {
 func (iter *ExprIter) Reset() (findMatch bool) {
 	defer func() { iter.matched = findMatch }()
 
+	if !iter.EngineTypeSet.Contain(iter.Group.EngineType) {
```

Comment:
Since the `GroupExpr`s in a Group share the same EngineType, we only need to check it here.

Reply:
please add this comment to the codebase.